lib/tasks/addon-install.js

```diff
@@ -42,6 +42,7 @@ class AddonInstallTask extends Task {
       'save': commandOptions.save,
       'save-dev': !commandOptions.save,
       'save-exact': commandOptions.saveExact,
+      'useYarn': commandOptions.yarn,
```

Comment:
@PieterJanVdb we're explicitly passing `commandOptions.yarn` to the `NpmInstallTask` constructor above. I'm not quite sure why we need to pass it here too 🤔

Reply:
It's because `#run` is checking `options.useYarn` here: https://github.com/ember-cli/ember-cli/blob/master/lib/tasks/npm-task.js#L139

Looks like we can just remove that line instead.

Reply:
Just tested it locally and it seems to work. Opened a quick PR here: https://github.com/ember-cli/ember-cli/pull/7481